api/graphql/types/reservations/serializers/staff_reservation_modify_serializers.py

```diff
@@ -145,22 +147,28 @@ def validate(self, data):
                 ValidationErrorCodes.RESERVATION_MODIFICATION_NOT_ALLOWED,
             )
 
-        begin = data.get("begin", self.instance.begin)
-        end = data.get("end", self.instance.end)
+        begin = data.get("begin", self.instance.begin).astimezone(DEFAULT_TIMEZONE)
+        end = data.get("end", self.instance.end).astimezone(DEFAULT_TIMEZONE)
 
-        new_buffer_before = data.get("buffer_time_before", None)
-        new_buffer_after = data.get("buffer_time_after", None)
+        new_buffer_before: datetime.timedelta | None = data.get("buffer_time_before", None)
+        new_buffer_after: datetime.timedelta | None = data.get("buffer_time_after", None)
 
+        reservation_unit: ReservationUnit
         for reservation_unit in self.instance.reservation_unit.all():
+            # Can't set buffers for whole day reservations
```

Comment:
`StaffReservationModify` seems to be wrong of old, here it is not be possible to try to change the time information of the reservation, just the other details, so the existing buffers should be always kept.
That is, time and buffers always through `StaffReservationAdjustTime`; so like `begin` and `end`, `buffer_time_before` and `buffer_time_after` should not be mutable here. (Consequently the `overlap/buffer_times/intervals` validations are bogus here, I guess.)

I suppose the reason for the old implementation might have been that the option was reserved that staff could adjust just the buffers without moving the reservation itself to another time, but it would be better to do that with `StaffReservationAdjustTime`, especially now that the setting `reservation_block_whole_day` may apply.

Reply:
Need to check with frontend that these can be removed, but otherwise, I'm for removing them.

Reply:
Asked Joonatan, and they are not used, so removing is safe.

Reply:
Actually, the field is read-only, so couldn't have been used to write, but the checks were still there for some reason... Removing them.

Reply:
_Actually_ actually, I just asked about the begin and end fields, mistakingly, and the buffers ARE mutated in the `staffReservationModify` endpoint. Removing them would require making 2 mutations with the current setup, so I suggest keeping this in until we can better refactor the reservation endpoint. @vergama